WPMedia/engine-theme-sdk
# TMEDIA-21 - Add ErrorBoundary component

## Description

Add ErrorBoundary component for handling failing children components and provide a fallback

As part of the work in TMEDIA-21 for lazy loading a block we have the need to have an error boundary to catch the failing component loading within an anonymous function property

## Jira Ticket
- [TMEDIA-21](https://arcpublishing.atlassian.net/browse/TMEDIA-21)

## Test Steps
1. View storybook
2. Verify tests

## Dependencies or Side Effects
Some block will become dependent on this component

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
